Backend/django_mobx_test/restaurant/views.py

```diff
@@ -0,0 +1,182 @@
+            ticket.save()
+            ticket_data = TicketModelSerializer(ticket)
+            return Response({'data': ticket_data.data}, status=status.HTTP_201_CREATED)
+
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    @action(detail=True, methods=['get'])
+    def tickets(self, request, pk=None):
+        try:
+            restaurant = Restaurant.objects.get(id=pk, owner=request.user)
+            tickets = Ticket.objects.filter(restaurant=restaurant)
+            ticket_data = TicketModelSerializer(tickets, many=True)
+            return Response({'data': ticket_data.data}, status=status.HTTP_200_OK)
+
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    @action(detail=True, methods=['delete'])
+    def ticket_id(self, request, pk=None):
+        try:
+            restaurant = Restaurant.objects.get(id=pk, owner=request.user)
+            tickets = Ticket.objects.get(id=request.query_params['id'], restaurant=restaurant)
+            tickets.delete()
+            return Response({'data': status.HTTP_200_OK}, status=status.HTTP_200_OK)
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    @action(detail=True, methods=['get'])
+    def specific_ticket(self, request, pk=None):
+        try:
+            restaurant = Restaurant.objects.get(id=pk, owner=request.user)
+            tickets = Ticket.objects.get(id=request.query_params['id'], restaurant=restaurant)
+            ticket_data = TicketModelSerializer(tickets)
+            return Response({'data': ticket_data.data}, status=status.HTTP_200_OK)
+
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+
+class PurchasedTicketViewSet(ViewSet):
+    permission_classes = (AllowAny, )
+
+    def list(self, request):
+        try:
+            tickets = Ticket.objects.all()
+            ticket_data = TicketModelSerializer(tickets, many=True)
+            return Response({'data': ticket_data.data}, status=status.HTTP_200_OK)
+
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    def retrieve(self, request, pk=None):
+        try:
+            ticket = Ticket.objects.get(ticket_coupon=pk)
+            ticket_data = TicketModelSerializer(ticket)
+            return Response({'data': ticket_data.data}, status=status.HTTP_200_OK)
+
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    def create(self, request):
+        try:
+            ticket = Ticket.objects.get(ticket_coupon=request.data['ticket_coupon'])
+            if ticket.max_purchase != ticket.sell_out:
```

Comment:
If the concurrency requests comes in at the same time in thread mode, it should not work, that's said the two requests will be on different thread and they got same result on retrieving ticket (not always but sometimes, so defined as concurrency).